### PR TITLE
[NO REVIEW] Do not assign internal key comparator to user comparator

### PR DESCRIPTION
**Pull request by riversand963**
Test plan:
make check

### CODE REVIEW

File: table/sst_file_dumper.cc [hunk 1]
```diff
@@ -125,7 +125,7 @@ Status SstFileDumper::GetTableReader(const std::string& file_path) {
                               nullptr);
       file_.reset(new RandomAccessFileReader(std::move(file), file_path));
     }
-    options_.comparator = &internal_comparator_;
+    options_.comparator = internal_comparator_.user_comparator();
```
Comment (satyajanga) [inline table/sst_file_dumper.cc:128]: where is the `options_.comparator ` referenced? Is it passed into table reader or sst file reader?

Comment (riversand963) [reply table/sst_file_dumper.cc:128]: It's like not passed. The subsequent `NewTableReader` call uses `ioptions_` which is constructed from `options_` and has a `comparator` member.

Comment (riversand963) [reply table/sst_file_dumper.cc:128]: Right. Therefore, I suspect it's not at all used. We should fix this, or at least make it clearer.

Comment (riversand963) [reply table/sst_file_dumper.cc:128]: @satyajanga to clarify, this PR is only intended for testing and I currently do not plan to merge it.
Feel free to discuss and address this issue in your PR.

**Pull request closed without merging**